src/main/java/mops/module/services/ModulWrapperService.java

```diff
@@ -0,0 +1,114 @@
+package mops.module.services;
+
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.Set;
+import mops.module.database.Modul;
+import mops.module.database.Veranstaltung;
+import mops.module.database.Veranstaltungsform;
+import mops.module.database.Zusatzfeld;
+import mops.module.wrapper.ModulWrapper;
+import org.springframework.stereotype.Service;
+
+@Service
+public class ModulWrapperService {
+
+    public static int VERANSTALTUNGSFORMEN_PRO_VERANSTALTUNG = 6;
+    public static int ZUSATZFELDER_PRO_VERANSTALTUNG = 2;
+
+    /**
+     * Entpackt aus einem ModulWrapper-Objekt ein Modul-Objekt.
+     * @param modulWrapper Im Formular empfangener ModulWrapper.
+     * @return Ein entsprechendes Modul-Objekt.
+     */
+    public static Modul readModulFromWrapper(ModulWrapper modulWrapper) {
+        for (int i = 0; i < modulWrapper.getVeranstaltungen().size(); i++) {
+            modulWrapper.getVeranstaltungen().get(i).setVeranstaltungsformen(
+                    new HashSet<>(modulWrapper.getVeranstaltungsformen()[i]));
+            modulWrapper.getVeranstaltungen().get(i).setZusatzfelder(
+                    new HashSet<>(modulWrapper.getZusatzfelder()[i]));
+        }
+        Set<Veranstaltung> veranstaltungenSet = new HashSet<>(modulWrapper.getVeranstaltungen());
+
+        Modul modul = modulWrapper.getModul();
+        modul.setVeranstaltungen(veranstaltungenSet);
+        modul.refreshMapping();
+        return modul;
+    }
+
+    /**
+     * Erstellt einen leeren ModulWrapper für ein noch nicht exisitierendes Modul, der entsprechende
+     * Listen von neuen Veranstaltung-, Veranstaltungsform- und Zusatzfeld-Instanzen enthält,
+     * damit über diese im Frontend iteriert und ihre Felder befüllt werden können.
+     * @param veranstaltungsanzahl Anzahl der Veranstaltungen in dem Modul.
+     * @return Das entsprechende ModulWrapper-Objekt.
+     */
+    @SuppressWarnings("unchecked")
+    public static ModulWrapper initializeEmptyWrapper(int veranstaltungsanzahl) {
+
+        Modul modul = new Modul();
+        ModulWrapper modulWrapper = new ModulWrapper(modul,
+                null, null, null);
+        modulWrapper.veranstaltungen = new LinkedList<>();
+        modulWrapper.veranstaltungsformen = new LinkedList[veranstaltungsanzahl];
+        modulWrapper.zusatzfelder = new LinkedList[veranstaltungsanzahl];
```

Comment:
Vorschlag: Statt Arrays von Listen wären auch Listen von Listen denkbar (als konkrete Implementierung dann zB ArrayList<>). Damit fällt auch die "unchecked" Warnung weg und der Code wird ein bisschen verboser (weil "add" und "get" statt diese eckigen Klammern...) Hab die Klasse in meiner lokalen Version geändert und kann euch das gerne zeigen.

Reply:
Datenstruktur wird durch "Modulbearbeitung" ggf. verändert.